### PR TITLE
Allowlist cco_credentials_mode metric from cloud credential operator.

### PR DESCRIPTION
**Pull request by dgoodwin**
This metric reports the mode the operator is configured to use, which we
would like to see across the entire fleet to understand real world
customer use.

* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by dgoodwin** [comment]
Results from this metric currently include what modes are disabled:

```

{mode="credsremoved",service="cco-metrics"} | 0
{mode="degraded",service="cco-metrics"} | 0
{mode="manual",service="cco-metrics"} | 0
{mode="mint",service="cco-metrics"} | 1
{mode="passthrough",service="cco-metrics"} | 0
{mode="unknown",service="cco-metrics"} | 0
```

This may need changing, please let me know. These 6 modes will likely be 8 or 10 within a few releases.

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -236,6 +236,8 @@ data:
     - '{__name__="che_workspace_start_time_seconds_sum"}'
     # (codeready workspaces, @ibuziuk) The overall number of attempts for starting all the workspaces.
     - '{__name__="che_workspace_start_time_seconds_count"}'
+    # (cloud credential operator, @dgoodwin) Track current mode the cloud-credentials-operator is functioning under.
```
Comment (s-urbaniak) [inline Documentation/data-collection.md:239]: please add a github team, not a single person. We have precedences, but they were exceptions which need to be resolved.

**Comment by s-urbaniak** [comment]
cc'ing @smarterclayton for final approval

File: Documentation/data-collection.md (hunk 1 above)
Comment (dgoodwin) [reply Documentation/data-collection.md:239]: My mistake, addressed now.

**Comment by dgoodwin** [comment]
Based on discussion on slack I think we need to modify this in the code to only report the 1's. Can I pursue this separately or does that have to be in place before this can merge?

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 2]
```diff
@@ -228,6 +228,8 @@ data:
     - '{__name__="che_workspace_start_time_seconds_sum"}'
     # (codeready workspaces, @ibuziuk) The overall number of attempts for starting all the workspaces.
     - '{__name__="che_workspace_start_time_seconds_count"}'
+    # (cloud credential operator, @openshift/openshift-team-hive) Track current mode the cloud-credentials-operator is functioning under.
```
Comment (lilic) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:231]: Is this operator installed by default or is an optional component?

Comment (dgoodwin) [reply manifests/0000_50_cluster-monitoring-operator_04-config.yaml:231]: Installed by default CVO operator.

**Comment by dgoodwin** [comment]
Removing the 0 metrics for 4.7: https://github.com/openshift/cloud-credential-operator/pull/256

**Comment by dgoodwin** [comment]
Bump, would like to get this wrapped up this sprint. Can anyone approve?

**Pull request merged**